collect_app/src/main/java/org/odk/collect/android/activities/WebViewActivity.java

```diff
@@ -0,0 +1,88 @@
+package org.odk.collect.android.activities;
+
+import android.graphics.Bitmap;
+import android.os.Bundle;
+import android.support.v7.app.AppCompatActivity;
+import android.support.v7.widget.Toolbar;
+import android.view.View;
+import android.webkit.WebResourceError;
+import android.webkit.WebResourceRequest;
+import android.webkit.WebView;
+import android.webkit.WebViewClient;
+import android.widget.ProgressBar;
+
+import org.odk.collect.android.R;
+
+/**
+ * Created by dexter on 29/1/18.
+ */
+
+public class WebViewActivity extends AppCompatActivity {
+
+    private WebView webView;
+    private ProgressBar progressBar;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_web_view);
+
+        Toolbar toolbar = (Toolbar) findViewById(R.id.toolbar);
+        setSupportActionBar(toolbar);
+        getSupportActionBar().setDisplayHomeAsUpEnabled(true);
+        getSupportActionBar().setHomeAsUpIndicator(R.drawable.ic_close);
+
+        String url = getIntent().getStringExtra("url");
```

Comment:
It would be good to add a static filed for this string then you can use such a field in both places (both places you use "url") which is safer.